js/songs/edit.controller.js

```diff
@@ -0,0 +1,21 @@
+"use strict";
+
+(function(){
+  angular
+  .module("songs")
+  .controller("SongEditController", [
+    "SongFactory",
+    "$stateParams",
+    SongEditControllerFunction
+  ]);
+
+  function SongEditControllerFunction(SongFactory, $stateParams) {
+    this.song = SongFactory.get({id: $stateParams.id});
```

Comment:
Once we loaded `$state` as a dependency, we can pass that into the controller function to use it.